blocks/library/paragraph/index.js

```diff
@@ -94,27 +98,69 @@ registerBlockType( 'core/paragraph', {
 		}
 	},
 
-	edit( { attributes, setAttributes, insertBlocksAfter, focus, setFocus, mergeBlocks, onReplace } ) {
-		const { align, content, dropCap, placeholder, fontSize, backgroundColor, textColor, width } = attributes;
-		const toggleDropCap = () => setAttributes( { dropCap: ! dropCap } );
-		const className = dropCap ? 'has-drop-cap' : null;
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
 
-		return [
-			focus && (
-				<BlockControls key="controls">
-					<AlignmentToolbar
-						value={ align }
-						onChange={ ( nextAlign ) => {
-							setAttributes( { align: nextAlign } );
-						} }
-					/>
-				</BlockControls>
-			),
-			focus && (
-				<InspectorControls key="inspector">
-					<BlockDescription>
-						<p>{ __( 'Text. Great things start here.' ) }</p>
-					</BlockDescription>
+			this.state = { activeFormats: { }, activeFormatSettings: null, footnoteId: null };
+
+			this.footnotes = this.props.attributes.footnotes || {};
+
+			this.footnoteFormatter = {
```

Comment:
This seems to require a significant amount of effort on the part of the block implementer to track the state of the active node in TinyMCE and apply formatting. 

If I understand correctly the needs around identifying and applying formats / tokens, there's a few things which would be ideal:

- A reasonable API in TinyMCE to define formats / token types, how they’re identified & applied, maybe emitting specific events when the selection becomes "active" for this type (to integrate with toolbar)
- Define a set of built-in / global formats: bold, italic, strikethrough, footnotes
   - Managed within Editable, not surfaced to block implementer
- If possible, expose a simple-to-use prop on Editable for custom formats

So extending the bold, italic idea, one could enable a footnotes format on Editable via `formattingControls={ [ 'bold', 'footnotes' ] }`.

It seems like there could be some overlap here with how TinyMCE enables toolbar buttons to be defined with a `stateSelector` property that handles much of the work around determining whether a button should be active or not.